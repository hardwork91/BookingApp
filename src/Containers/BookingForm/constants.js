export const TOOLTIP_DELAY = 1;
export const ORIGIN_SELECTED_DATA = "originSelectedData";
export const DESTINATION_SELECTED_DATA = "destinationSelectedData";

export const ROUND_TRIP = "Round-trip";
export const ONE_WAY = "One-way";
export const MULTI_CITY = "Multi-city";
export const TRIP_BUILDER = "Trip Builder";

export const FLIGHT_TYPES = [ROUND_TRIP, ONE_WAY, MULTI_CITY, TRIP_BUILDER];

export const ECONOMY = "Economy";
export const BUSINESS = "Business";
export const FIRST = "First";

export const TRAVEL_CLASES = [ECONOMY, BUSINESS, FIRST];
