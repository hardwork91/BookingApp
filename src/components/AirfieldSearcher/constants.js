export const MAX_AIRPORT_ALLOWED = 3;

export const SEARCH_AIRFIELD_REQUEST = "SEARCH_AIRFIELD_REQUEST";
export const SEARCH_AIRFIELD_SUCCESS = "SEARCH_AIRFIELD_SUCCESS";
export const SEARCH_AIRFIELD_FAILURE = "SEARCH_AIRFIELD_FAILURE";

export const CLEAR_AIRFIELD_REQUEST = "CLEAR_AIRFIELD_REQUEST";
