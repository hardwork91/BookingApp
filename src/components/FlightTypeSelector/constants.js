export const ROUND_TRIP = "Round-trip";
export const ONE_WAY = "One-way";
export const MULTI_CITY = "Multi-city";
export const TRIP_BUILDER = "Trip Builder";

export const FLIGHT_TYPES = [ROUND_TRIP, ONE_WAY, MULTI_CITY, TRIP_BUILDER];
