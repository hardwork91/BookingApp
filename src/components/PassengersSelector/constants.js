// All categories are exported individually in case of being needed
// Each gategory has Name(used as label and key) and Nescription
const CHILD = "Child";
export const CHILD_DESCRIPTION = "- 11";
const YOUTH = "Youth";
export const YOUTH_DESCRIPTION = "12 - 17";
const ADULTS = "Adults";
export const ADULTS_DESCRIPTION = "18 - 64";
const SENIORS = "Seniors";
export const SENIORS_DESCRIPTION = "65 +";

// Add a new category is easy, just need to add it on the PASSENGERS_CATEGORIES array
export const PASSENGERS_CATEGORIES = [CHILD, YOUTH, ADULTS, SENIORS];

// Itinal values also easy to modify
export const PASSENGERS_INITIAL_VALUE = {
  [CHILD]: 0,
  [YOUTH]: 0,
  [ADULTS]: 1,
  [SENIORS]: 0,
};
