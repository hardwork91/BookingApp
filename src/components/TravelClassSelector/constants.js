export const ECONOMY = "Economy";
export const BUSINESS = "Business";
export const FIRST = "First";

export const TRAVEL_CLASES = [ECONOMY, BUSINESS, FIRST];
