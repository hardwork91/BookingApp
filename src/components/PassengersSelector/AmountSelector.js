import React, { Component } from "react";
import PropTypes from "prop-types";
import { InputNumber, Typography } from "antd";
import * as constants from "./constants";

const { Text } = Typography;

export default class AmountSelector extends Component {
  // update the passenger selector state when change the spinner value
  handleChange = value => {
    const { category, onChange } = this.props;
    onChange(category, value);
  };

  render() {
    const { category, amount } = this.props;
    return (
      <div>
        <Text style={{ marginRight: ".5em" }}>{category}</Text>
        <Text type="secondary">
          {constants[`${category.toUpperCase()}_DESCRIPTION`]}
        </Text>
        <InputNumber
          style={{ float: "right", marginLeft: "1em" }}
          size="small"
          // imput number min value is setted to 0
          min={0}
          defaultValue={amount}
          onChange={this.handleChange}
          precision={0}
        />
      </div>
    );
  }
}

// prop types validation
AmountSelector.propTypes = {
  category: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  amount: PropTypes.number.isRequired,
};
