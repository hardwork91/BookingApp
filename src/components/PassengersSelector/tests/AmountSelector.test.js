import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { mount, configure } from "enzyme";
import toJson from "enzyme-to-json";
import AmountSelector from "../AmountSelector";
import { CHILD } from "../constants";

configure({ adapter: new Adapter() });

const setup = props => ({
  category: CHILD,
  onChange: jest.fn(),
  amount: 2,
  ...props,
});

describe("AmountSelector", () => {
  it("render as expected", () => {
    const tree = mount(<AmountSelector {...setup()} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("when handleChange", () => {
    const props = setup();
    const tree = mount(<AmountSelector {...props} />);
    const value = 4;
    tree.instance().handleChange(value);
    expect(props.onChange).toBeCalledWith(CHILD, value);
  });
});
