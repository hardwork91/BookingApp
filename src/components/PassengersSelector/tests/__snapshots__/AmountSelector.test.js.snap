// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountSelector render as expected 1`] = `
<AmountSelector
  amount={2}
  category="Child"
  onChange={[Function]}
>
  <div>
    <Text
      style={
        Object {
          "marginRight": ".5em",
        }
      }
    >
      <withConfigConsumer(Base)
        component="span"
        ellipsis={false}
        style={
          Object {
            "marginRight": ".5em",
          }
        }
      >
        <Base
          component="span"
          ellipsis={false}
          getPrefixCls={[Function]}
          prefixCls="ant-typography"
          renderEmpty={[Function]}
          style={
            Object {
              "marginRight": ".5em",
            }
          }
        >
          <LocaleReceiver
            componentName="Text"
          >
            <ResizeObserver
              disabled={true}
              onResize={[Function]}
            >
              <Typography
                aria-label={null}
                className=""
                component="span"
                style={
                  Object {
                    "WebkitLineClamp": null,
                    "marginRight": ".5em",
                  }
                }
              >
                <span
                  aria-label={null}
                  className="ant-typography"
                  style={
                    Object {
                      "WebkitLineClamp": null,
                      "marginRight": ".5em",
                    }
                  }
                >
                  Child
                </span>
              </Typography>
            </ResizeObserver>
          </LocaleReceiver>
        </Base>
      </withConfigConsumer(Base)>
    </Text>
    <Text
      type="secondary"
    >
      <withConfigConsumer(Base)
        component="span"
        ellipsis={false}
        type="secondary"
      >
        <Base
          component="span"
          ellipsis={false}
          getPrefixCls={[Function]}
          prefixCls="ant-typography"
          renderEmpty={[Function]}
          type="secondary"
        >
          <LocaleReceiver
            componentName="Text"
          >
            <ResizeObserver
              disabled={true}
              onResize={[Function]}
            >
              <Typography
                aria-label={null}
                className="ant-typography-secondary"
                component="span"
                style={
                  Object {
                    "WebkitLineClamp": null,
                  }
                }
              >
                <span
                  aria-label={null}
                  className="ant-typography ant-typography-secondary"
                  style={
                    Object {
                      "WebkitLineClamp": null,
                    }
                  }
                >
                  - 11
                </span>
              </Typography>
            </ResizeObserver>
          </LocaleReceiver>
        </Base>
      </withConfigConsumer(Base)>
    </Text>
    <InputNumber
      defaultValue={2}
      min={0}
      onChange={[Function]}
      precision={0}
      size="small"
      step={1}
      style={
        Object {
          "float": "right",
          "marginLeft": "1em",
        }
      }
    >
      <InputNumber
        autoComplete="off"
        className="ant-input-number-sm"
        defaultValue={2}
        downHandler={
          <Icon
            className="ant-input-number-handler-down-inner"
            type="down"
          />
        }
        focusOnUpDown={true}
        min={0}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onPressEnter={[Function]}
        parser={[Function]}
        precision={0}
        prefixCls="ant-input-number"
        required={false}
        step={1}
        style={
          Object {
            "float": "right",
            "marginLeft": "1em",
          }
        }
        upHandler={
          <Icon
            className="ant-input-number-handler-up-inner"
            type="up"
          />
        }
        useTouch={false}
      >
        <div
          className="ant-input-number ant-input-number-sm"
          style={
            Object {
              "float": "right",
              "marginLeft": "1em",
            }
          }
        >
          <div
            className="ant-input-number-handler-wrap"
          >
            <InputHandler
              aria-disabled={false}
              aria-label="Increase Value"
              className="ant-input-number-handler ant-input-number-handler-up "
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              prefixCls="ant-input-number"
              role="button"
              unselectable="unselectable"
            >
              <TouchFeedback
                activeClassName="ant-input-number-handler-active"
                disabled={false}
              >
                <span
                  aria-disabled={false}
                  aria-label="Increase Value"
                  className="ant-input-number-handler ant-input-number-handler-up "
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  unselectable="unselectable"
                >
                  <Icon
                    className="ant-input-number-handler-up-inner"
                    type="up"
                  >
                    <LocaleReceiver
                      componentName="Icon"
                    >
                      <i
                        aria-label="icon: up"
                        className="anticon anticon-up ant-input-number-handler-up-inner"
                      >
                        <IconReact
                          className=""
                          type="up-o"
                        >
                          <svg
                            aria-hidden="true"
                            className=""
                            data-icon="up"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            key="svg-up"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                              key="svg-up-svg-0"
                            />
                          </svg>
                        </IconReact>
                      </i>
                    </LocaleReceiver>
                  </Icon>
                </span>
              </TouchFeedback>
            </InputHandler>
            <InputHandler
              aria-disabled={false}
              aria-label="Decrease Value"
              className="ant-input-number-handler ant-input-number-handler-down "
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              prefixCls="ant-input-number"
              role="button"
              unselectable="unselectable"
            >
              <TouchFeedback
                activeClassName="ant-input-number-handler-active"
                disabled={false}
              >
                <span
                  aria-disabled={false}
                  aria-label="Decrease Value"
                  className="ant-input-number-handler ant-input-number-handler-down "
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  unselectable="unselectable"
                >
                  <Icon
                    className="ant-input-number-handler-down-inner"
                    type="down"
                  >
                    <LocaleReceiver
                      componentName="Icon"
                    >
                      <i
                        aria-label="icon: down"
                        className="anticon anticon-down ant-input-number-handler-down-inner"
                      >
                        <IconReact
                          className=""
                          type="down-o"
                        >
                          <svg
                            aria-hidden="true"
                            className=""
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            key="svg-down"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              key="svg-down-svg-0"
                            />
                          </svg>
                        </IconReact>
                      </i>
                    </LocaleReceiver>
                  </Icon>
                </span>
              </TouchFeedback>
            </InputHandler>
          </div>
          <div
            className="ant-input-number-input-wrap"
          >
            <input
              aria-valuemin={0}
              aria-valuenow={2}
              autoComplete="off"
              className="ant-input-number-input"
              min={0}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              required={false}
              role="spinbutton"
              step={1}
              value="2"
            />
          </div>
        </div>
      </InputNumber>
    </InputNumber>
  </div>
</AmountSelector>
`;
